clippy_lints/src/casts/mod.rs

```diff
@@ -506,6 +507,33 @@ declare_clippy_lint! {
     "casting the result of `abs()` to an unsigned integer can panic"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for a raw slice being cast to a slice pointer.
+    ///
+    /// ### Why is this bad?
+    /// This can result in multiple `&mut` references to the same location when only a pointer is
+    /// required.
+    /// `ptr::slice_from_raw_parts` is a safe alternative that doesn't require
+    /// the same [safety requirements] to be upheld.
+    ///
+    /// ### Example
+    /// ```rust,ignore
+    /// let _: *const [u8] = std::slice::from_raw_parts(ptr, len) as *const _;
+    /// let _: *mut [u8] = std::slice::from_raw_parts_mut(ptr, len) as *mut _;
+    /// ```
+    /// Use instead:
+    /// ```rust,ignore
+    /// let _: *const [u8] = std::ptr::slice_from_raw_parts(ptr, len);
+    /// let _: *mut [u8] = std::ptr::slice_from_raw_parts_mut(ptr, len);
+    /// ```
+    /// [safety requirements]: https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html#safety
+    #[clippy::version = "1.64.0"]
+    pub RAW_SLICE_POINTER_CAST,
```

Comment:
The lint name should start with `cast_` for consistency with (most of) the other cast lints. Maybe `cast_raw_parts_slice` or `cast_from_raw_parts_slice`? To me `raw slice` sounds like it refers to something specific rather than a regular slice constructed via `from_raw_parts`

Reply:
`cast_slice_from_raw_parts` would be a good name I think, it starts with cast to highlight the undesirable action, and includes the fact that it's a slice reference made from the `slice::from_raw_parts` function.  One problem with that name is that `ptr::slice_from_raw_parts` is similarly named, perhaps `cast_slice_reference_from_raw_parts` to highlight that it's a slice reference being cast specifically?

Reply:
Sorry for the delay; I brought it up [in zulip](https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/Lint.20name.20for.20clippy.239247) a couple days ago. I think `cast_slice_from_raw_parts` would be better @clubby789

With that + squashing the commits I think this is ready to merge